### PR TITLE
Wizard recipe: Pythia8-v8.244.0

### PR DESCRIPTION
**Pull request by jstrube**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: Pythia8
* Version: v8.244.0

@staticfloat please review and merge.

### CODE REVIEW

File: P/Pythia8/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+sources = [
+    "http://home.thep.lu.se/~torbjorn/pythia8/pythia8244.tgz" =>
+    "e34880f999daf19cdd893a187123927ba77d1bf851e30f6ea9ec89591f4c92ca",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd pythia8244/
+./configure --prefix=${prefix} --enable-shared --enable-64bit
```
Comment (giordano) [inline P/Pythia8/build_tarballs.jl]: Aren't you missing the configuration of the host (quite important for cross-compilations)?
```suggestion
./configure --prefix=${prefix} --host=${target} --enable-shared --enable-64bit
```

File: P/Pythia8/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+products = [
+    LibraryProduct("libpythia8", :libpythia8)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```
Comment (giordano) [inline P/Pythia8/build_tarballs.jl]: Is GCC 9 really required by this library?  Keep in mind https://github.com/JuliaPackaging/Yggdrasil/wiki/Build-Tricks#building-with-an-old-gcc-version-a-library-that-has-dependencies-built-with-newer-gcc-versions when you choose to use recent GCC versions.

File: P/Pythia8/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Pythia8"
```
Comment (giordano) [inline P/Pythia8/build_tarballs.jl]: It looks like the name of the library is "PYTHIA", not "Pythia8", we try to keep the same name as the upstream project, including capitalisation.

File: P/Pythia8/build_tarballs.jl (hunk 1 above)
Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: Yeah, there's no --host on the configure script. I only get "unknown option --host" do you have a suggestion?

File: P/Pythia8/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:x86_64, libc=:glibc),
+    Linux(:powerpc64le, libc=:glibc),
+    MacOS(:x86_64)
+]
+
```
Comment (giordano) [inline P/Pythia8/build_tarballs.jl]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1081&view=logs&j=2b277407-6ab0-5e15-108f-8beae357ef17&t=551d77a4-d348-5c2d-a045-3b696b2c9100&l=1803
```
[ Info: /workspace/P/Pythia8/build/powerpc64le-linux-gnu/bbE9s0Ue/destdir/lib/libpythia8.so locks us to cxx11
┌ Warning: /workspace/P/Pythia8/build/powerpc64le-linux-gnu/bbE9s0Ue/destdir/lib/libpythia8.so contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder ~/.julia/dev/BinaryBuilder/src/auditor/compiler_abi.jl:189
```
```suggestion
platforms = expand_cxxstring_abis(platforms)
```

File: P/Pythia8/build_tarballs.jl (hunk 3 above)
Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: I'm happy to change, but Pythia 6 is all Fortran (hence the caps), while Pythia8 is all C++. I think maybe somebody would like both. Do you have a suggestion?

File: P/Pythia8/build_tarballs.jl (hunk 2 above)
Comment (giordano) [reply P/Pythia8/build_tarballs.jl]: [This](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1081&view=logs&j=0f152180-077b-50d8-339d-4d860b71551b&t=99fec6b7-45c0-5795-d783-6a06c518257a&l=1811) is the problem with compiling with a too recent version of GCC: the library can't just be dlopen'ed by many systems:
```
ERROR: could not load library "/workspace/P/Pythia8/build/x86_64-linux-gnu/qEoTbMrR/destdir/lib/libpythia8.so"
/usr/local/julia/bin/../lib/julia/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /workspace/P/Pythia8/build/x86_64-linux-gnu/qEoTbMrR/destdir/lib/libpythia8.so)
[...]
[ Info: Found a valid dl path libpythia8.so while looking for libpythia8
[ Info: /workspace/P/Pythia8/build/x86_64-linux-gnu/qEoTbMrR/destdir/lib/libpythia8.so matches our search criteria of libpythia8
[ Info: /workspace/P/Pythia8/build/x86_64-linux-gnu/qEoTbMrR/destdir/lib/libpythia8.so cannot be dlopen'ed
```

You have to choose the oldest version as possible, compatibly with the requirements of the library of course.

Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: Could probably be removed... Thanks for checking.

File: P/Pythia8/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply P/Pythia8/build_tarballs.jl]: Oh, I see: this is _not_ using autotools, yet they called a script with the same name, similar options, but not all of them.  They got the bonus points from section VII of https://archive.fosdem.org/2018/schedule/event/how_to_make_package_managers_cry/.  Great job!

File: P/Pythia8/build_tarballs.jl (hunk 3 above)
Comment (giordano) [reply P/Pythia8/build_tarballs.jl]: We can have multiple version of the same package at the same time, see e.g. [OpenBLAS](https://github.com/JuliaPackaging/Yggdrasil/tree/a43603d4e548800b57ff3f4513eabdc10a0dc4f2/O/OpenBLAS) or [LLVM](https://github.com/JuliaPackaging/Yggdrasil/tree/a43603d4e548800b57ff3f4513eabdc10a0dc4f2/L/LLVM).

File: P/Pythia8/build_tarballs.jl (hunk 1 above)
Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: Yeah, particle physicists and code. If it compiles, it's probably OK...

File: P/Pythia8/build_tarballs.jl (hunk 3 above)
Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: OK. Should I close this and put it into the right dir structure?

Comment (giordano) [reply P/Pythia8/build_tarballs.jl]: Why need to close?

Comment (jstrube) [reply P/Pythia8/build_tarballs.jl]: Ah. Right. I have a terminal and don't need to go through the web interface :-)

**Pull request merged**